File: examples/perl/download_metagenome_sequences.pl
#!/usr/bin/env perl
use JSON;
use LWP::UserAgent;
use URI::Escape;
use Data::Dumper;

# get a user agent
my $ua = LWP::UserAgent->new;
$ua->agent("MyClient/0.1 ");

my $key = $ENV{"MGRKEY"};
# define the parameters
my $metagenome = $ARGV[0] ; 
my $group_level = "family" ;
my $result_type = "abundance";
my $source = "SEED";

die "Metagenome ID argument is required!   
Usage:    download_metagenome_sequences.pl <mgr accession number>   
Example:  download_metagenome_sequences.pl mgm4440442.5\n "  unless $#ARGV + 1 == 1;
die "Don't recognize format of $metagenome" unless $metagenome =~ m/\d\d\d\d\d\d\d.\d/;

my $response;
my $content;
my $file_suffix = '.fastq.gz';
my $filename;
# retrieve the data
foreach my $file_id ('050.1', '050.2') {
	my $base_url = "http://api.metagenomics.anl.gov/1/download/$metagenome?file=".$file_id;
	my $url = $base_url."&auth=$key";
	print STDERR "Retrieving $url\n";
	
	if ($file_id eq '050.2') {
		$file_suffix = '.fasta.gz';
	}
	$filename = "$metagenome".$file_suffix;
	$response = $ua->get($url, ":content_file" =>  $filename);
	# check http header 
	$content = $response->content;
	if ($response->is_success) {
		last;
	}
}
die "Error with HTTP request:  ". $response->status_line."\n".$content unless $response->is_success ;
print STDERR "Writing \"$filename\"\n";

File: examples/python/download_metagenome_sequences.py
#!/usr/bin/env python

# This script calls the MG-RAST API and attempts to download the raw data file for a
# metagenome whose MG-RAST ID is specified.

import sys
import os
import re
from optparse import OptionParser
from mglib.mglib import get_auth_token, file_from_url, API_URL

def retrieveMGRbyaccession(accessionno, key):
    '''Retrieve raw data from MG-RAST API using curl and dump result into file named <accession>.gz'''
    try:
        a = re.search(r"(\d\d\d\d\d\d\d\.\d)$", accessionno).group(1)
    except IndexError:
        sys.exit("Don't recognize accession number format %s" % accessionno)
    if key == "":
        s1 = "%s/download/mgm%s?file=050.1" % (API_URL, a)
    else:
        s1 = "%s/download/mgm%s?file=050.1&auth=%s" % (API_URL, a, key)
    sys.stderr.write("Retrieving %s\n" % s1)
    with open("%s.gz" % a, "w") as FH:
        file_from_url(s1, FH)

if __name__ == '__main__':
    usage  = '''usage: download_metagenome_sequences.py <accession number>
example: download_metagenome_sequences.py  MGR4440613.3'''
    parser = OptionParser(usage)
    (opts, args) = parser.parse_args()
# Assign the value of key from the OS environment
    key = get_auth_token()
    if len(key) == 25:
        sys.stderr.write("Using MGR webkey %s\n" % key)
    else:
        sys.stderr.write("Warning: MGR webkey not defined\n")
# test for correct number of arguments
    try:
        accession = args[0]
    except IndexError:
        parser.error("accession is a required parameter\n")

# call retrieveMGRbyaccession
    retrieveMGRbyaccession(accession, key)
